### PR TITLE
Added phpdocs

### PR DESCRIPTION
**Pull request by staabm**
Increase phpstan level

### CODE REVIEW

**Comment by staabm** [comment]
Opened a upstream issue regarding the false positive

> Method Seld\JsonLint\JsonParser::lint() should return Seld\JsonLint\ParsingException|null but return statement is missing.

https://github.com/phpstan/phpstan/issues/3738

File: src/Seld/JsonLint/JsonParser.php [hunk 1]
```diff
@@ -326,6 +332,7 @@ public function parse($input, $flags = 0)
             }
         }
 
+        // @phpstan-ignore-next-line
```
Comment (Seldaek) [inline src/Seld/JsonLint/JsonParser.php:335]: Why is this needed? Is true not part of mixed?!

File: phpstan-config.neon [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
-    level: 1
+    level: 5
 
     paths:
         - src/
         - tests/
+
+    ignoreErrors:
+        -
+           message: '#Method Seld\JsonLint\JsonParser::lint\(\) should return Seld\JsonLint\ParsingException|null but return statement is missing.#'
```
Comment (stof) [inline phpstan-config.neon:10]: I would add the `return null;` instead of ignoring the error.

File: phpstan-config.neon [hunk 3]
```diff
@@ -1,6 +1,11 @@
 parameters:
-    level: 1
+    level: 5
```
Comment (stof) [inline phpstan-config.neon:2]: is there anything blocking from using higher levels ?

File: src/Seld/JsonLint/JsonParser.php (hunk 1 above)
Comment (staabm) [reply src/Seld/JsonLint/JsonParser.php:335]: Phpstan thinks this is unreachable code

File: phpstan-config.neon (hunk 3 above)
Comment (staabm) [reply phpstan-config.neon:2]: Nope, I have choosen this more or less random and started fixing.

Increasing it further could be subject for a new PR

File: phpstan-config.neon (hunk 2 above)
Comment (staabm) [reply phpstan-config.neon]: fixed

File: src/Seld/JsonLint/JsonParser.php (hunk 1 above)
Comment (Seldaek) [reply src/Seld/JsonLint/JsonParser.php:336]: I guess technically it is, there doesn't seem to be a way out of the while (true) above, except for the returns, so I don't think this is needed.

Comment (staabm) [reply src/Seld/JsonLint/JsonParser.php:336]: so would you agree we throw a exception in case this path is taken?

Comment (Seldaek) [reply src/Seld/JsonLint/JsonParser.php:336]: Yeah sounds good, if the tests pass :D

Comment (stof) [reply src/Seld/JsonLint/JsonParser.php:336]: or just remove the statement entirely, if this path cannot be taken anyway.

File: phpstan-config.neon (hunk 3 above)
Comment (stof) [reply phpstan-config.neon:2]: well, I would suggest to try running it at max level (`8`) and see whether it finds anything else. If it is already green at max level with the changes done here, there is no reason to wait for another PR.

File: src/Seld/JsonLint/JsonParser.php (hunk 1 above)
Comment (staabm) [reply src/Seld/JsonLint/JsonParser.php]: phpstan also complained about the `throw Exception` line to be never reachable.. just deleted it, as suggested by stof.

File: phpstan-config.neon (hunk 3 above)
Comment (staabm) [reply phpstan-config.neon:2]: justed tested level max and level 6.

as of level 6 phpstan requires all properties to be typed.

its not so much work but I dont want to invest it now.

**Comment by Seldaek** [comment]
OK I think good to merge now? :)

**Comment by staabm** [comment]
> OK I think good to merge now? :)

Yep
